spek-ide-plugin/intellij-common/src/main/kotlin/org/spekframework/intellij/base.kt

```diff
@@ -92,12 +99,29 @@ abstract class SpekBaseRunConfiguration<T: RunConfigurationModule>(name: String,
             "($it)"
         } ?: ""
 
-        return if (path.name.isEmpty()) {
-            "$prefix Spek tests in <default>"
+        return if (path.name.isEmpty() && parent != null && parent.isRoot) {
+            "$prefix Spek tests in <default package>"
         } else if (parent != null && parent.isRoot) {
             "$prefix Spek tests ${path.name}"
         } else {
             "$prefix ${path.name}"
         }.trim()
     }
 }
+
+object SpekScopeLocator: SMTestLocator {
+    override fun getLocation(protocol: String, path: String, project: Project, scope: GlobalSearchScope): MutableList<Location<PsiElement>> {
+        if (protocol != "spek") {
+            return mutableListOf()
+        }
+        val locations = mutableListOf<Location<PsiElement>>()
+        val path = PathBuilder.parse(path).build()
+        val descriptor = ScopeDescriptorCache.findDescriptor(path)
+
+        if (descriptor != null) {
+            locations.add(PsiLocation(descriptor.element))
```

Comment:
not a fan of MutableLists. Would just change it to

```kotlin
return if (descriptor != null) listOf(PsiLocation(descriptor.element)) else emptyList
```

Reply:
also avoids allocation in empty case that way

Reply:
TIL: `emptyList()` 👍